static/js/compose_actions.js

```diff
@@ -243,6 +243,39 @@ exports.cancel = function () {
     $(document).trigger($.Event('compose_canceled.zulip'));
 };
 
+exports.respond_to_empty = function (opts) {
+    if (current_msg_list.selected_message()) {
+        // If message exists, use respond_to_message instead.
+        exports.respond_to_message(opts);
```

Comment:
This seems unnecessary to me and sort of dangerous, as if the conditional were off, you'd get infinite recursion between the two functions.